### PR TITLE
Added "Final" flag for Catching URIs and removed "Default" flag for Surfaces

### PR DESCRIPTION
**Pull request by un-tone**
For https://github.com/StarcounterPrefabs/Website/issues/41

### CODE REVIEW

**Comment by un-tone** [comment]
Related to the PR https://github.com/StarcounterPrefabs/Simplified/pull/45

File: src/Website/Helpers/DataHelper.cs [hunk 1]
```diff
@@ -56,14 +54,20 @@ public void GenerateDefaultSurface()
                 Default = true
             };
 
+            var catchAllUrl = new WebUrl()
+            {
+                Template = surface,
+                Url = null,
+                IsFinal = true
+            };
```
Comment (warpech) [inline src/Website/Helpers/DataHelper.cs]: I think that a similar Catch All + IsFinal rule should be added to `GenerateAppHubSurface` and `GenerateLauncherSurface`

From the current catching rules, only SidebarSurface rule for `/website/cms/surfaces` is not final.

File: src/WebsiteProvider/Api/ContentHandlers.cs [hunk 2]
```diff
@@ -82,26 +81,18 @@ public void Register()
...
                 WebUrl webUrl = Db.SQL<WebUrl>("SELECT wu FROM Simplified.Ring6.WebUrl wu WHERE wu.Url = ?", requestUri).First;
 
                 if (webUrl == null)
                 {
                     string wildCard = GetWildCardUrl(requestUri);
 
-                    webUrl = Db.SQL<WebUrl>("SELECT wu FROM Simplified.Ring6.WebUrl wu WHERE wu.Url = ?", wildCard).First;
+                    webUrl = Db.SQL<WebUrl>("SELECT wu FROM Simplified.Ring6.WebUrl wu WHERE wu.Url = ?", wildCard).First
+                             ?? Db.SQL<WebUrl>("SELECT wu FROM Simplified.Ring6.WebUrl wu WHERE wu.Url IS NULL AND wu.IsFinal = ?", true).First
+                             ?? Db.SQL<WebUrl>("SELECT wu FROM Simplified.Ring6.WebUrl wu WHERE wu.Url IS NULL").First;
```
Comment (warpech) [inline src/WebsiteProvider/Api/ContentHandlers.cs]: Will this catch a manually created catch-all rule? I think the `Url` value will be an empty string, so the last two SQLs should check for null or empty string.

**Comment by warpech** [comment]
@un-tone can you also please update README.md to be relevant after this change?

**Comment by un-tone** [comment]
@warpech could you clarify what info do you want to be added to the README.md exactly? I'm not confident since it contains only publishing instructions currently.

File: src/Website/Helpers/DataHelper.cs (hunk 1 above)
Comment (un-tone) [inline src/Website/Helpers/DataHelper.cs]: Set IsFinal flag for default ApHubTemplate catching rules

File: src/WebsiteProvider/Api/ContentHandlers.cs (hunk 2 above)
Comment (un-tone) [inline src/WebsiteProvider/Api/ContentHandlers.cs]: Fixed the conditions.

**Comment by warpech** [comment]
I mean README.md in the `develop`. You will know when you see it, just search for the word "Default".

Anyway. I will merge `develop` to `master` now to avoid confusion.

**Comment by warpech** [comment]
Perfect explanation in README.md, thanks

**Pull request merged**